### DISCUSSION TITLE
Toxicity overhead removal

### DISCUSSION THREAD

**Discussion by Berberbrand** [runelite/runelite#16591] [Ideas]
So I pvp and I deal with a lot of toxicity.

The entity hider has "hide other 2D" which removes all overhead chat, it would be nice to have the ability to choose what people can say around you. A small box underneath the option, maybe a sub category box where you could type "exceptions" would be nice. The only reason why is because when you turn on hide all others 2D it hides "Taste vengeance!" and it makes it difficult to fight.

I figured either a new plugin could be made easily using the same code or just making an exception box.

**Comment by Nightfirecat** [comment]
Use the Chat Filter plugin

**Reply by Berberbrand** [reply]
The chat filter plugin does not have an option for -All- and to make * exceptions to certain words.

**Reply by Nightfirecat** [reply]
You can write a regex which matches everything except certain terms. https://stackoverflow.com/questions/2404010/match-everything-except-for-specified-strings

**Reply by Berberbrand** [reply]
You can write a ragex in the "filtered ragex" part of the plugin? Im not really understanding how you could write the code properly and the exceptions if so.

**Reply by Berberbrand** [reply]
This is what ive got so far and this blocks out all chat
([A-Za-z0-9\-\_\:\)]+)
But after that how would I create an exception to a phrase?

**Reply by Nightfirecat** [reply]
Per the link in my previous comment, a regex such as `^(?!.*(red|green|blue))` should filter all messages except those with "red", "green", or "blue" in them.

**Reply by Berberbrand** [reply]
I got you, it didn't seem to work with me. I will try it again maybe I can get somewhere

**Reply by Berberbrand** [reply]
Tysm man, I think I got it working. I thought you would have to create a code then another one to make the exception. I had tried that one code before but I had the other line as well.
